### PR TITLE
42 creating get profile

### PR DESCRIPTION
**Pull request by IvanKhudyakov**
Trello task https://trello.com/c/2vxb4fQ3/42-creating-get-profile

2 new routes with authentication middleware:
- to get the profile details (name, email and id)
- to update the user name and\or email

### CODE REVIEW

File: src/controllers/userController.ts [hunk 1]
```diff
@@ -62,4 +63,75 @@ const login = async (
...
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+
+    const { email } = req.body;
```
Comment (JaneGH) [inline src/controllers/userController.ts]: const email = req.query.email as string; With GET requests, parameters are passed in the URL, so it’s commonly done with query parameters and rewrite Swagger after this

Comment (chukwuemeka-fincra) [inline src/controllers/userController.ts]: if user is authenticated then you have this in the user variable on line 72, 
you don't need to use any input from req.body, that is a vulnerability

File: src/controllers/userController.ts [hunk 2]
```diff
@@ -62,4 +63,75 @@ const login = async (
...
+  const { body: { name, email }, params: { id: userIdToBeUpdated } } = req;
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  //depending on the Front we can change the approach and compare by e.g. email
+  if (user.userId.toString() !== userIdToBeUpdated) {
```
Comment (chukwuemeka-fincra) [inline src/controllers/userController.ts]: same comment as above, if the authenticated profile is been updated, then use the identifier from the authenticated user when updating the db

File: src/controllers/userController.ts [hunk 3]
```diff
@@ -62,4 +63,75 @@ const login = async (
...
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+
+    const { email } = req.body;
+
+    if (!email) {
```
Comment (JaneGH) [inline src/controllers/userController.ts]: i am not shure that we need email here, we could rewrite like this: const email = req.query.email as string;

    if (email) {
        userProfile = await User.findOne({ email: email });
    }else{ 
        userProfile = await User.findOne({ _id: user.userId });
    }

**Review by JaneGH — CHANGES_REQUESTED** [review]
good job, need a little changes

File: src/controllers/userController.ts (hunk 1 above)
Comment (IvanKhudyakov) [reply src/controllers/userController.ts]: fixed. new reques should look like htttp://localhost:8000/api/v1/auth/profile?email=ivan@ivan.c1om

Comment (IvanKhudyakov) [reply src/controllers/userController.ts]: fixed by using query parameter for cases where user will request profile of another user

File: src/controllers/userController.ts (hunk 3 above)
Comment (IvanKhudyakov) [reply src/controllers/userController.ts]: Fixed

File: src/controllers/userController.ts (hunk 2 above)
Comment (IvanKhudyakov) [reply src/controllers/userController.ts]: Fixed. Now the only option to update a profile is to update your own profile, no way to update another profile

**Review by JaneGH — APPROVED** [review]

**Review by JaneGH — APPROVED** [review]

**Review by JaneGH — APPROVED** [review]

**Pull request merged**